app/models/manageiq/providers/redhat/infra_manager/refresh/refresher.rb

```diff
@@ -34,6 +34,8 @@ def collect_inventory_for_targets(ems, targets)
           end
         end
 
+        empty_hash_when_target_not_found!(target, data)
```

Comment:
How that would work when a `target` is ems?

Reply:
`corresponding_hashes_empty?` should take care of skipping the ems, but that can be more explicit